core/src/main/java/com/yahoo/oak/ScopedReadBuffer.java

```diff
@@ -123,14 +109,23 @@ public double getDouble(int index) {
     }
 
     /**
-     * Allows access to the memory address of the underlying off-heap ByteBuffer of Oak.
+     * Allows access to the memory address of the OakUnsafeDirectBuffer of Oak.
      * The address will point to the beginning of the user data, but avoiding overflow is the developer responsibility.
      * Thus, the developer should use getLength() and access data only in this boundary.
-     * This is equivalent to ((DirectBuffer) b.getByteBuffer()).address() + b.getOffset()
      *
-     * @return the exact memory address of the underlying buffer in the position of the data.
+     * @return the exact memory address of the Buffer in the position of the data.
      */
-    @Override public long getAddress() {
+    @Override public long getAddress() { //for both OakUnsafeDirectBuffer -- OakScopedReadBuffer
         return s.getAddress();
     }
+    
+   //must never be called internally kept in here to beautify the interface
```

Comment:
It should be called internally, but externally it can be called. So it is not just for the beauty of the interface...

Reply:
in this case it is, we throw exception here

Reply:
lets discuss